digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -339,3 +339,73 @@ pub async fn get_by_id(
         groups: grouping,
     })
 }
+
+pub async fn get_signatures_for_asset(
+    conn: &impl ConnectionTrait,
+    asset_id: Option<Vec<u8>>,
+    tree_id: Option<Vec<u8>>,
+    leaf_idx: Option<i64>,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<(String, Option<String>)>, DbErr> {
+    // if tree_id and leaf_idx are provided, use them directly to fetch transactions
+    if let (Some(tree_id), Some(leaf_idx)) = (tree_id, leaf_idx) {
+        let transactions = fetch_transactions(conn, tree_id, leaf_idx, pagination, limit).await?;
+        return Ok(transactions);
+    }
+
+    if asset_id.is_none() {
+        return Err(DbErr::Custom(
+            "Either 'id' or both 'tree' and 'leafIndex' must be provided".to_string(),
+        ));
+    }
+
+    // if only asset_id is provided, fetch the latest tree and leaf_idx (asset.nonce) for the asset
+    // and use them to fetch transactions
+    let stmt = asset::Entity::find()
+        .distinct_on([(asset::Entity, asset::Column::Id)])
+        .filter(asset::Column::Id.eq(asset_id))
+        .order_by(asset::Column::Id, Order::Desc)
+        .limit(1);
+    let asset = stmt.one(conn).await?;
+    if let Some(asset) = asset {
+        let tree = asset
+            .tree_id
+            .ok_or(DbErr::RecordNotFound("Tree not found".to_string()))?;
+        if tree.is_empty() {
+            return Err(DbErr::Custom("Empty tree for asset".to_string()));
+        }
+        let leaf_id = asset
+            .nonce
+            .ok_or(DbErr::RecordNotFound("Leaf ID does not exist".to_string()))?;
+        let transactions = fetch_transactions(conn, tree, leaf_id, pagination, limit).await?;
+        Ok(transactions)
+    } else {
+        Ok(Vec::new())
+    }
+}
+
+pub async fn fetch_transactions(
+    conn: &impl ConnectionTrait,
+    tree: Vec<u8>,
+    leaf_id: i64,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<(String, Option<String>)>, DbErr> {
+    let mut stmt = cl_audits::Entity::find().filter(cl_audits::Column::Tree.eq(tree));
```

Comment:
Just overall thinking about the `cl_audits` table, which probably contains every tree that is indexed.  Won't this get pretty big and need to be purged sometimes like maybe after a tree is fully audited?  If that is the case do we want a public API method that depends on it?  Or should we just stick with stuff within the tools/ directory?

Also, another thing is that `cl_audits` table is configurable, so, for example, `insert_change_log`  takes a bool indicating whether to populate that table.  I think we need a similar check so that if `cl_items` is not present in the configuration, it returns some kind of "not available in this system" type of messsage.  I didn't see that anywhere, please correct me if I missed it.

Reply:
I think that makes sense. The provider can decide whether they enable it or not. A user could also enable it for a subset of trees.